polus-cellpose-inference-plugin/src/main.py

```diff
@@ -0,0 +1,346 @@
+                                                               x_overlap:x_max - x_min + x_overlap,
+                                                               ...].transpose(4,3,2,0,1)
+        root[0:1,1:3,z:z + 1,y_min:y_max,x_min:x_max] = dP[:,
+                                                           y_overlap:y_max - y_min + y_overlap,
+                                                           x_overlap:x_max - x_min + x_overlap,
+                                                           ...].transpose(4,0,3,1,2)
+        
+    return True
+
+def main(inpDir: Path,
+         pretrained_model: str,
+         diameterMode: str,
+         diameter: float,
+         filePattern: str,
+         outDir: Path):
+    
+    """ Sanity check on diameter mode """
+    assert diameterMode in ['Manual','PixelSize','FirstImage','EveryImage']
+    
+    # Get all file names in inpDir image collection
+    fp = filepattern.FilePattern(inpDir,filePattern)
+    inpDir_files = [f[0]['file'] for f in fp]
+
+    # Error checking for diameterMode==Manual
+    if diameterMode=='Manual' and not diameter:
+        logger.warning('Manual diameter selection specified, but manual diameter is 0 or None. Using Cellpose model defaults.')
+        
+        try:
+            d = DEV.get()
+            model = models.Cellpose(model_type=pretrained_model,
+                                    gpu=USE_GPU,
+                                    device=d)
+            DEV.put(d)
+            diameter = model.diam_mean
+            
+        except Exception as err:
+            logger.error('Manual diameter selected, but not diameter supplied and model has no default diameter.')
+            raise err
+        
+    elif diameterMode=='EveryImage':
+        diameter = 0
+    
+    logger.info(f'Running {DEV.qsize()} workers')
+    
+    # Initialize the process pool
+    if DEV.qsize() == 1:
+        initialize_model(DEV,pretrained_model)
+        executor = ThreadPoolExecutor(1)
+    else:
+        executor = ProcessPoolExecutor(DEV.qsize(),
+                                       initializer=initialize_model,
+                                       initargs=(DEV,pretrained_model))
+    
+    if diameterMode == 'FirstImage':
+        
+        logger.info('Estimating diameter from the first image...')
+        
+        process = executor.submit(estimate_diameter,inpDir_files[0])
+        
+        diameter = process.result()
+        
+    # Loop through files in inpDir image collection and process
+    processes = []
+    for f in inpDir_files:
+        br = BioReader(f.absolute())
+        out_file = Path(outDir).joinpath(f.name.replace('.ome','_flow.ome').replace('.tif','.zarr')).absolute()
+        bw = BioWriter(out_file,metadata=br.metadata)
+        bw.C = 4
+        bw.dtype = np.float32
+        bw.channel_names = ['cell_probability','x','y','labels']
+        
+        bw._backend._init_writer()
```

Comment:
Why is it necessary to call the backend? Would be useful to know why this programming pattern was used with biowriter